src/Rules/Methods/CallToMethodStatementWithoutSideEffectsRule.php

```diff
@@ -61,8 +62,8 @@ public function processNode(Node $node, Scope $scope): array
 		}
 
 		$method = $calledOnType->getMethod($methodName, $scope);
-		if ($method->hasSideEffects()->no() || $node->expr->isFirstClassCallable()) {
-			if (!$node->expr->isFirstClassCallable()) {
+		if ($method->hasSideEffects()->no() || $methodCall->isFirstClassCallable()) {
```

Comment:
You don't need to check `$method->hasSideEffects()`, that's already implied by NoopExpressionNode.

Reply:
And you don't need to check `instanceof NeverType` on line 74 either, that's also implied.